### ISSUE TITLE
Reshape conversion produces invalid output shape (if one dim is -1)

### ISSUE THREAD

**Issue by lvadke** [tensorflow/tensorflow#52028]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04.4 LTS
- TensorFlow installation (pip package or built from source): pip package
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.6.0

### 2. Code

```
import numpy as np
import tensorflow as tf

inputs = tf.keras.Input(shape=(1, 126))
outputs = tf.keras.layers.Reshape((-1, 21))(inputs)
#outputs = tf.reshape(inputs, (-1, 21))
model = tf.keras.Model(inputs=inputs, outputs=outputs)

model.summary()

model.compile(optimizer='sgd', loss='mean_squared_error')

inputs = np.random.rand(126).reshape((1, 1, 126))
outputs = np.random.rand(126).reshape((1, 6, 21))

model.fit(x=inputs, y=outputs, epochs=1)

def representative_dataset_gen():
  for input in inputs:
    input = input.astype(np.float32)
    yield [input]

converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.representative_dataset = representative_dataset_gen

tflite_model = converter.convert()
with open('model.tflite', 'wb') as f:
  f.write(tflite_model)
```

### 3. Failure after conversion
Conversion produces a reshape operation (and all subsequent ops) with incorrect output shape, which makes TFLM fail (as it uses only the output shape as the reshape input parameter):
![reshape](https://user-images.githubusercontent.com/52713197/133601423-e585f707-953e-4b20-9374-7453a5becd54.png)
I tried both, tf.keras.layers.Reshape and tf.reshape, and the result is the same, except the branch in the blue box is generated only for the Keras operator. (The issue is extracted from a detection model with many reshapes.)

**Comment by mohantym** [comment]
Hi @ymodak ,Could you please look at this issue ,Providing GIST for reference. in TF [2.5](https://colab.research.google.com/gist/mohantym/7c9682a7405c7ff8c544db1883ec2cfe/github_52028.ipynb#scrollTo=1mn-mL9rZUUQ),[2.6](https://colab.research.google.com/gist/mohantym/de0b92956820c5b1f1125923e408ec8f/github_52028_2-6.ipynb),[2.7](https://colab.research.google.com/gist/mohantym/9f6c61db5e2653d8ecfda0ffc4bb3eff/github_52028.ipynb#scrollTo=1mn-mL9rZUUQ)

**Comment by karimnosseir** [comment]
Sorry for the late reply.
This looks correct, the shape gets visualized is the default shape where any dynamic dimension uses '1'. If you printed the input/output details you will see that the output has shape_signature = [-1, -1, 21]
